src/libs/SearchUtils.ts

```diff
@@ -76,9 +77,39 @@ function getShouldShowMerchant(data: OnyxTypes.SearchResults['data']): boolean {
     });
 }
 
+function doesAtleastOneExpenseBelongToAPastYear(data: TransactionListItemType[] | ReportListItemType[] | OnyxTypes.SearchResults['data']): boolean {
+    if (Array.isArray(data)) {
+        return data.some((item: TransactionListItemType | ReportListItemType) => {
+            if ('transactions' in item) {
+                // If the item is a ReportListItemType, iterate over its transactions and check them
+                return item.transactions.some((transaction) => {
+                    const transactionYear = new Date(transaction?.modifiedCreated ? transaction.modifiedCreated : transaction?.created || '').getFullYear();
+                    return transactionYear !== new Date().getFullYear();
+                });
+            }
+
+            const createdYear = new Date(item?.modifiedCreated ? item.modifiedCreated : item?.created || '').getFullYear();
```

Comment:
https://github.com/Expensify/App/blob/95c8202a0e8fdf133fc65bd8d249c7d91372e2ff/src/libs/TransactionUtils.ts#L465

Reply:
Yep, since this logic is reused it would be great to abstract it